### PR TITLE
Group 2: Barely get it to work on time

### PR DESCRIPTION
**Pull request by TehMillhouse**

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+    }
+
+    private List<String> parseLine(String line) {
+        return Arrays.asList(line.split(","));
+    }
+    
+    private void publishCsvFile(CsvFile csvFile) {
+        for (String caption : csvFile.headers) {
+            this.csvUi.addColumn(caption);
+        }
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:49]: "column" "caption" and "header" is used mainly for the same thing. Maybe it's not that good.

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+
+    public void run(final File csvFile) throws Exception {
+        this.csvFile = parseFile(csvFile);
+        publishCsvFile(this.csvFile);
+    }
+
+    private CsvFile parseFile(final File csvFile) throws FileNotFoundException {
+        BufferedReader input = new BufferedReader(new FileReader(csvFile));
+
+        String headerLine = input.lines().findFirst().get();
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:34]: This gloriously crashes on empty files

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 3]
```diff
@@ -1,17 +1,57 @@
...
+        List<String> headers = parseLine(headerLine);
+
+        List<List<String>> contents = input.lines().skip(1).map(line -> parseLine(line)).collect(Collectors.toList());
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:37]: This line is excessively long

File: src/main/java/de/cas/mse/exercise/csv/CsvReader.java [hunk 4]
```diff
@@ -1,17 +1,57 @@
...
 
-	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
-	}
-	
+    private CsvUi csvUi;
+    private CsvFile csvFile;
+
+    public List<List<String>> filter(int col, String word) {
+        return csvFile.contents.stream().filter(row -> row.get(col).contains(word)).collect(Collectors.toList());
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/csv/CsvReader.java:22]: This line is excessively long, also it'd be nicer if `CsvFile` implements the filtering itself

Comment (zopad) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:22]: We have almost the exact same impl :D

Comment (TehMillhouse) [reply src/main/java/de/cas/mse/exercise/csv/CsvReader.java:22]: My point stands! :D

**Pull request merged**